### ISSUE TITLE
Screen reader does not announce when hovering over drop down

### ISSUE THREAD

**Issue by stevencl** [Microsoft/vscode#42476]
### Issue Type
Bug

### Description

Enable screen reader
Open output panel
Hover over controls from right to left, starting with top right
Hear the screen reader announce each control "Close panel", "Move to right", "Toggle maximized panel", "Toggle output scroll lock", "Clear output" but nothing when hovering over the drop down

Expected: Screen reader should announce when hovering over the drop down

### VS Code Info

VS Code version: Code - Insiders 1.20.0-insider (eed7e193125f3d4cfebe2089d1b2a9c674388754, 2018-01-30T09:51:29.494Z)
OS version: Windows_NT x64 10.0.16299

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-3687U CPU @ 2.10GHz (4 x 2594)|
|Memory (System)|7.89GB (1.45GB free)|
|Process Argv|C:\Program Files\Microsoft VS Code Insiders\Code - Insiders.exe|
|Screen Reader|yes|
|VM|0%|

</details><details><summary>Extensions (12)</summary>

Extension|Author (truncated)|Version
---|---|---
html-snippets|abu|0.2.1
vscode-eslint|dba|1.4.4
tslint|eg2|1.0.24
vscode-great-icons|emm|2.1.22
vscode-gutter-preview|kis|0.12.2
csharp|ms-|1.13.1
vs-keybindings|ms-|0.1.7
vsliveshare|ms-|0.1.0-preview.171
team|ms-|1.122.0
debugger-for-chrome|msj|4.1.0
addDocComments|ste|0.0.8
cordova-tools|vsm|1.3.3

</details>
Reproduces only with extensions

 Found when testing: https://github.com/Microsoft/vscode/issues/40589

**Comment by bpasero** [comment]
@stevencl good catch!

@cleidigh fyi

**Comment by bpasero** [comment]
@stevencl can you clarify what you mean by "hover", are you using keyboard to move focus around the controls? I am not hearing any feedback when just using the mouse to hover. Also, is the dropdown expanded or not when you try?

If the dropdown is not expanded then this s unrelated to the changes we did because the custom dropdown only kicks in when you expand the dropdown.

**Comment by stevencl** [comment]
Yes, hover with the mouse. I didn't try with the keyboard. Dropdown was not expanded.

**Comment by bpasero** [comment]
Ok, this did not change, so I suspect an upstream issue.

**Comment by bpasero** [comment]
@cleidigh I would expect that no ARIA role is required given that this is a pretty standard HTML control that a screen reader should be able to announce, no?

**Comment by cleidigh** [comment]
Triage...

@bpasero 
Don't want to pretend to be an expert but, as far as I can see from the specs as well as the issue thread below, this is more of a design decision to either have readable tooltips or live ARIA content.  I don't think there's any technical issue , including Electron, preventing us from doing this.  I'm not sure it makes sense though.  There are plenty of other examples of elements that could have this but, again I don't think it makes sense.

Consider closing?  

https://github.com/nvaccess/nvda/issues/6048

**Comment by bpasero** [comment]
@cleidigh are you suggesting to have a `aria-label` role or `title` attribute on the select element?

**Comment by cleidigh** [comment]
I just realized several of the clients for selectbox do not pass in a label, outputs and terminal for example.
ironically I just posted two issues for this earlier this evening.  Maybe some readers can be configured to read on  mouse hover press. 
I will check tomorrow

**Comment by bpasero** [comment]
@isidorn can we ask our accessibility experts what the expectations are? Should a screen reader announce a dropdown when hovering over?

**Comment by Neurrone** [comment]
@isidorn what drop down is this?

**Comment by cleidigh** [comment]
@stevencl 
If the standard NVDA supports it presumably by reading the main label, then it certainly makes sense.  I was just raising concerns about some of the oddities mentioned in the NVDA issue thread.
I need to see this working as right now I do not get output from mouse hovers but there may be an issue with my use of Dragon for both mouse and keyboard.  I will find a helper to do some more testing.

This directly dovetails with my efforts to make the selectbox a well implemented accessible widget.  

@isidorn this affects any drop-down.

BTW I have a couple of issues opened on just the related main labels.
#60724 
#60726

**Comment by cleidigh** [comment]
@stevencl 
I got a little bit of testing done.  Trying several recent versions of Code and NVDA 2018.2 under Windows 10, I have not been able to hear output on the panel buttons.  I did find an option under mouse configuration for speaking tooltips on hover.  That option is disabled by default for the current version.  When enabled I get output like:

"Close Panel tooltip"

Your example output mentioned above does not include "tooltip".  Unless you just omitted that, I assume I instill seeing something different.  With or without this option enabled, I do hear output on text items.

Any thoughts on configuration issues since I cannot reproduce this?

**Comment by cleidigh** [comment]
This still may be an upstream issue with Chromium/Electron.  
I discovered that at some point after my July 5 Commit that added
label support for selectbox , things regressed and this label
no longer works in 1.28.  Below is the setup that initially worked.
This was just before moving to Electron 2.x

@bpasero pointed me to the 28.2 Exploration version.  I just tested
in that version the new labeling works again.  I still cannot
hear output over the panel buttons, but the issues may be related.

I'm thinking that there may be a 2.x Electron that somehow allowed
NVDA to announce the buttons and then regressed similarly to my issues
with the main selectbox label.

Version: 1.26.0
Commit: Unknown
Date: Unknown
Electron: 1.7.12
Chrome: 58.0.3029.110
Node.js: 7.9.0
V8: 5.8.283.38
Architecture: x64

**Comment by cleidigh** [comment]
@stevencl 
I am still unable to reproduce reader output hovering with mouse over panel buttons.  Is it possible your
NVDA is set up to read tooltips?

**Comment by stevencl** [comment]
Sorry for the slow response. I've uploaded a video demonstrating this: https://flic.kr/p/2cp7bBv

This is with NVDA set up not to announce tooltips.

**Comment by cleidigh** [comment]
@stevencl 
Great, thanks.
Can you let me know what version of NVDA you are using?
I still cannot reproduce the output you hear.  There are multiple
issues related to the Electron version that could be part of this and the problem is the version you tested
was 1.x now we have 2.12 I believe and test with Electron 3 all show slightly different ARIA oddities.

That said, until I can reproduce your baseline, it's hard to approach this.  I will keep on trying.

**Comment by stevencl** [comment]
My NVDA version is 2018.3.2

I just installed today's VSC insiders release and the screen reader now announces when hovering over the drop down. The video I shared yesterday is from the latest VSC stable release.

Thanks for persevering with this @cleidigh

**Comment by cleidigh** [comment]
@stevencl 
Excellent ! I just wish I could still reproduce this.  I did Discover that Dragon NaturallySpeaking interferes with NVDA.  Actually it's more correct to state that the reader does not recognize the virtual mouse of Dragon.  I bribed my daughter to use the mouse directly and found this out.  

 I still need to figure out why I do not The same output especially because this means I cannot reliability testing on some of the main issues I am working on.

I'd like to keep this open for that reason.  @bpasero can you get the correct output with your NVDA?
BTW:  I still believe this type of output with default reader set up is reading text as opposed to the elements specifically.

**Comment by bpasero** [comment]
@stevencl @cleidigh I am also on NVDA 2018.3.2 but I neither see any output when hovering over the dropdown, nor when hovering over the actions next to it (in the output panel). I am inside a VM though, so that could have easily an impact.

**Comment by cleidigh** [comment]
@stevencl  @bpasero 
I am on a native Windows 10 set up with identical behavior as @bpasero 's VM ,  I am doubtful the VM is an issue.  
@stevencl if you are still up for pursuing this, I would like to try to resolve the issue.
Perhaps you could export your configuration, maybe were missing something there.

**Comment by stevencl** [comment]
Sorry again for the delay.

@cleidigh - what configuration do you want me to export?

**Comment by cleidigh** [comment]
@stevencl 
The NVDA configuration, I cannot really use a perfectly   default configuration myself an adjustment to make sure we somehow have some little difference causing our different results.

**Comment by stevencl** [comment]
Here is the contents of my NVDA.ini file. Hope this is useful. Happy to get on a call with you too so you can inspect some more in detail.

schemaVersion = 2
[upgrade]
[update]
	autoCheck = True
	startupNotification = True
	allowUsageStats = True
	askedAllowUsageStats = True
[general]
	language = Windows
	saveConfigurationOnExit = True
	askToExit = True
	playStartAndExitSounds = True
	loggingLevel = INFO
	showWelcomeDialogAtStartup = True
[speech]
	synth = oneCore
	[[oneCore]]
		voice = HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Speech_OneCore\Voices\Tokens\MSTTS_V110_enUS_DavidM
		volume = 100
[presentation]
	reportTooltips = False
[braille]
	[[noBraille]]
		port = ""
[uwpOcr]
	language = en-US

**Comment by cleidigh** [comment]
@stevencl 
Thanks.  Certainly don't see anything suspicious as you alluded prior.  I will try to do some more testing and perhaps as you say a phone call might be good just  to noodle.
